### PR TITLE
Make Travis test that docs have been added to the branch

### LINKED ISSUE #34: Make sure all PRs are accompanied by docs
While working on GWM I keep finding outdated or incomplete docs. I spoke with @jordane on how to fix this and joked about making travis check that changes were made to the docs. He recommended that I actually do that.

### PR DESCRIPTION
**Pull request by iankronquist**
Fixes #34
Branched off of travis_ci_update
@tschuy r?

### CODE REVIEW

**Comment by tschuy** [comment]
There are no docs included. I have no idea what to do now because you didn't add docs about it :(

**Comment by tschuy** [comment]
But seriously, not sure I like this idea. not *everything* needs docs. I think it should be up to humans to keep it documented.

@Kennric what do you think?

**Comment by iankronquist** [comment]
Sure, I understand that. If you don't think docs are necessary you can argue it in the PR and we can merge without regard to Travis. Most new PRs and features *should* be accompanied by docs -- lacking docs is the exception, not the rule.
Also, the lack of docs is a deliberate test of the system.

**Comment by tschuy** [comment]
Also, I'd really appreciate it if you (a) branched off of travis-ci-update and then requested a merge into it, or (b) didn't call this branch "travis checks docs/whatever"... 

Perhaps rearrange the tests so the docs tests fail quickly? That way we don't run into waiting three minutes for travis to run the test suite while we could be working on docs.

**Comment by iankronquist** [comment]
Okay, I thought that the travis-ci-update branch would get merged before this branch (that one's simple, this one's going to have debate.) Let me know if I should close this and reopen it to merge into a different branch.

Lucy said she would prefer that the doc tests fail last, especially in the unusual case where that we don't have docs in a PR. I tend to agree, but if you'd prefer to fail fast I can change it.

The branch naming scheme is based off of the one in the [dev workflow](http://wiki.osuosl.org/development_process/1_development.html#the-development-process)
If you've got a beef with that, I would recommend you open an issue over there.

**Comment by tschuy** [comment]
I believe if they're tests travis would run them all anyway. why would you prefer to have them last?

I see about the branch name. Not a fan but don't dislike it enough to submit a PR.

**Comment by iankronquist** [comment]
@tschuy the more I think about it the less difference it seems to make. I changed it to the way you suggested.

**Comment by tschuy** [comment]
I'm going to look into having these appear as multiple different test suites so we can see if tests pass and docs failed, or whatever.

**Comment by iankronquist** [comment]
[It works!](https://travis-ci.org/osuosl/ganeti_webmgr/builds/63269969#L1005)

**Comment by tschuy** [comment]
Why are you matching three hashes?

**Comment by iankronquist** [comment]
@tschuy Thanks for asking. If HEAD was a merge commit, `git rev-list` gives us
`HEAD PARENT1 PARENT2`

File: scripts/travis-check-docs.bash [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# If Travis changes the way it starts up this script might break. Don't
+# despair! You'll probably only need to change PARENT_1 and PARENT_2 below,
+# everything else should work fine.
+#   $ git clone --depth=50 git://github.com/osuosl/ganeti_webmgr.git osuosl/ganeti_webmgr
+#   $ cd osuosl/ganeti_webmgr
+#   $ git fetch origin +refs/pull/36/merge:
+#   $ git checkout -qf FETCH_HEAD
+
+PREV_COMMIT=`git rev-list --parents HEAD -n 1`
+if [[ `echo $PREV_COMMIT | wc -w`  =~ '3' ]]; then
```
Comment (jordane) [inline scripts/travis-check-docs.bash]: You should check for `<=`, not `=~`. You can have more than two parents in a merge commit, though it is rare in our workflow.

Comment (iankronquist) [inline scripts/travis-check-docs.bash]: Aren't these things strings?

Comment (jordane) [inline scripts/travis-check-docs.bash]: This is bash, they're never *not* strings:

```
jme@frea> [[ '4' -ge 3 ]] && echo 'true'
true
```

**Comment by iankronquist** [comment]
Hurray, it works! @tschuy @jordane r?
It looks like one of our tests fails intermittently. That belongs in another issue.

**Comment by tschuy** [comment]
lgtm :+1: 

Perhaps you should squash commits so we don't get seven commits in a row of you fooling around with bash syntax?

**Comment by jordane** [comment]
> Perhaps you should squash commits so we don't get seven commits in a row of you fooling around with bash syntax?

Yes, please squash this.

**Comment by tschuy** [comment]
How far back should it be squashed? the first seven commits are from the branch this is branched from, but the others are fair game.

**Comment by jordane** [comment]
Certainly the last 7 commits of bash syntax should be squished together. 223f1de should also be squashed, as well as 8b16de7.

**Comment by iankronquist** [comment]
Squashed and rebased off of develop

**Comment by tschuy** [comment]
The only reason either of those failed was the boot_order test. 

+1

**Comment by iankronquist** [comment]
Damn that test. I'll look into it tomorrow.

**Comment by jordane** [comment]
+1

**Comment by tschuy** [comment]
We should merge https://github.com/osuosl/ganeti_webmgr/pull/30 first.

**Comment by iankronquist** [comment]
I'm going to rebase off develop now that #30 has been merged.

**Comment by tschuy** [comment]
lgtm :+1:

**Comment by iankronquist** [comment]
The PR CI behaves as expected, but the boot_order test fails.

**Comment by iankronquist** [comment]
The Push CI run behaves as expected. Merging.

**Pull request merged**